src/clj/puppetlabs/http/client/async.clj

```diff
@@ -293,14 +294,36 @@
         :else
           (DefaultRedirectStrategy.))))
 
+(schema/defn request-config :- RequestConfig
```

Comment:
This looks like a complete reimplementation of the Java version. Any reason not to directly reuse it?